src/init/starship.fish

```diff
@@ -1,19 +1,32 @@
+function __starship_set_job_count --description 'Set STARSHIP_JOBS using fish job groups (or legacy PIDs if toggled)'
+    # To force legacy behavior (process PIDs), set this variable to "false":
+    #   set -g __starship_fish_use_job_groups "false"
+    if test "$__starship_fish_use_job_groups" = "false"
+        # Legacy behavior: counts PIDs (may overcount pipelines with terminated producers)
+        set -l __count (jobs -p 2>/dev/null | count)
+    else
+        # Default behavior: count job groups
+        set -l __count (jobs -g 2>/dev/null | count)
+    end
+    set -gx STARSHIP_JOBS $__count
+end
+
 function fish_prompt
     switch "$fish_key_bindings"
         case fish_hybrid_key_bindings fish_vi_key_bindings
             set STARSHIP_KEYMAP "$fish_bind_mode"
         case '*'
             set STARSHIP_KEYMAP insert
     end
+
+    __starship_set_job_count
+
```

Comment:
I haven't tested this yet, but please but `__starship_set_job_count` where the previous `set STARSHIP_JOBS` line was; I am not certain, but calling it first may change `$status`.